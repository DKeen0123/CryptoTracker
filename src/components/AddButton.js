import React from 'react';

const AddButton = props => {
  return <button>{props.label}</button>;
};

export default AddButton;