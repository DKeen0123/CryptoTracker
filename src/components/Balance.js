import React from 'react';

const Balance = () => {
  return (
    <div>
      <h4>Your balance: </h4>
    </div>
  );
};

export default Balance;
