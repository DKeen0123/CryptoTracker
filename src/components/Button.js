import React from 'react';

const Button = () => {
  return <button>Deposit</button>;
};

export default Button;
