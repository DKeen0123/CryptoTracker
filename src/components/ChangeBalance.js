import React from 'react';

const ChangeBalance = () => {
  return (
    <div>
      <input />
    </div>
  );
};

export default ChangeBalance;
