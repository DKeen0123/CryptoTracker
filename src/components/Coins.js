import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../actions/action_bitcoin';
import { fetchEthereum } from '../actions/action_ethereum';

export class Coins extends Component {
  componentDidMount() {
    this.props.fetchBitcoin();
    this.props.fetchEthereum();
  }

  calculateBitcoin() {
    const { bitcoin } = this.props;
    if (Object.keys(bitcoin).length === 0) return '';
    return this.props.balance / parseInt(bitcoin[0].price_usd, 10);
  }

  calculateEthereum() {
    const { ethereum } = this.props;
    if (ethereum === null || ethereum === undefined) return '';
    if (Object.keys(ethereum).length === 0) return '';
    return this.props.balance / parseInt(ethereum[0].price_usd, 10);
  }

  render() {
    return (
      <div>
        <h3 className="bitcoin-balance">Bitcoin: {this.calculateBitcoin()}</h3>
        <h3 className="ethereum-balance">
          Ethereum: {this.calculateEthereum()}
        </h3>
      </div>
    );
  }
}

export default connect(state => state, { fetchBitcoin, fetchEthereum })(Coins);
