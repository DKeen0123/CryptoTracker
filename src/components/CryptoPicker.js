import React, { Component } from 'react';

class CryptoPicker extends Component {
  constructor(props) {
    super()
  }
  populateCryptos() {
    const { cryptos } = this.props;
    if (cryptos === null || cryptos === undefined) return '';
    if (Object.keys(cryptos).length === 0) return '';
    return cryptos.map(crypto => {
      return (
        <option id={crypto.id} key={crypto.id}>
          {crypto.name}
        </option>
      );
    });
  }

  render() {
    return (
      <div>
        <h3>Choose your crypto: </h3>
        <select
          onChange={this.props.selectCrypto}
          className="drop-down"
          value={this.props.chosenCrypto}
        >
          {this.populateCryptos()}
        </select>

      </div>
    );
  }
}

export default CryptoPicker;