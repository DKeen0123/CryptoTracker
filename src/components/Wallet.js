import React, { Component } from 'react';
import { connect } from 'react-redux';

export class Wallet extends Component {
  constructor() {
    super();

    this.state = { balance: undefined };
  }

  updateBalance = event =>
    this.setState({ balance: parseInt(event.target.value, 10) });

  render() {
    return (
      <div>
        <h3 className="balance">Balance: {this.props.balance}</h3>
        <input className="input-balance" onChange={this.updateBalance} />
      </div>
    );
  }
}

export default connect(state => {
  return { balance: state.balance };
}, null)(Wallet);
