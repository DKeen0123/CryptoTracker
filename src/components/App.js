import React from 'react';
import Wallet from './Wallet';
import Coins from './Coins';
import CryptoPicker from './CryptoPicker';

const App = () => {
  return (
    <div>
      <h2>CryptoTracker</h2>
      <hr />
      <Wallet />
      <CryptoPicker />
      <Coins />
    </div>
  );
};

export default App;
