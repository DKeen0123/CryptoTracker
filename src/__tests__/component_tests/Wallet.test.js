import React from 'react';
import { shallow, mount } from 'enzyme';
import { Wallet } from '../../components/Wallet';
describe('Wallet', () => {
  const mockDeposit = jest.fn();
  const mockWithdrawal = jest.fn();
  let props = { balance: 10, deposit: mockDeposit, withdraw: mockWithdrawal };
  let wallet = shallow(<Wallet {...props} />);

  it('renders correctly', () => {
    expect(wallet).toMatchSnapshot();
  });

  it('renders a Balance component', () => {
    expect(wallet.find('Balance').exists()).toBe(true);
  });

  it('renders a ChangeBalance component', () => {
    expect(wallet.find('ChangeBalance').exists()).toBe(true);
  });

  it('renders a CryptoPicker component', () => {
    expect(wallet.find('CryptoPicker').exists()).toBe(true);
  });

  describe('passing Props', () => {
    it('passes props.balance down to Balance component', () => {
      expect(wallet.find('Balance').prop('userBalance')).toEqual(10);
    });
    it('passes handleUpdateBalance() down to ChangeBalance component', () => {
      expect(wallet.find('ChangeBalance').prop('changeBalance')).toEqual(
        wallet.instance().handleUpdateBalance
      );
    });
  });

  describe('when mounted', () => {
    it('dispatches the `fetchCryptoNames()` method received from props', () => {
      const mockFetchCryptoNames = jest.fn();
      props.fetchCryptoNames = mockFetchCryptoNames;
      wallet = mount(<Wallet {...props} />);
      expect(mockFetchCryptoNames).toHaveBeenCalled();
    });
  });

  describe('handleUpdateBalance()', () => {
    it('sets the `currentInput` state to the amount entered into the input', () => {
      const event = { target: { value: 10 } };
      wallet.instance().handleUpdateBalance(event);
      expect(wallet.state('currentInput')).toEqual(10);
    });
  });

  describe('when a user types into .input-balance', () => {
    const userBalance = 25;
    beforeEach(() => {
      wallet
        .find('.input-balance')
        .simulate('change', { target: { value: userBalance } });
    });
    //
    // it('updates the local `state` balance and converts to a number', () => {
    //   expect(wallet.state().balance).toEqual(parseInt(userBalance, 10));
    // });

    describe('and the user wants to deposit into the balance', () => {
      beforeEach(() => {
        wallet.find('.deposit-btn').simulate('click');
      });

      it('dispatches the `deposit()` it receives from props with the local balance', () => {
        expect(mockDeposit).toHaveBeenCalledWith(parseInt(userBalance, 10));
      });
    });

    describe('and the user wants to withdraw from the balance', () => {
      beforeEach(() => {
        wallet.find('.withdrawal-btn').simulate('click');
      });

      it('dispatches the `withdraw()` it receives from props with the local balance', () => {
        expect(mockWithdrawal).toHaveBeenCalledWith(parseInt(userBalance, 10));
      });
    });
  });
});
