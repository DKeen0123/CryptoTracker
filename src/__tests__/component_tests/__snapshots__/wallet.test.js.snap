// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={10}
    deposit={[Function]}
    fetchCryptoNames={[Function]}
    withdraw={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Balance
          userBalance={10}
/>,
        <ChangeBalance
          changeBalance={[Function]}
/>,
        <Button
          className="btn-deposit"
          label="Deposit"
          transaction={[Function]}
/>,
        <Button
          className="btn-withdraw"
          label="Withdraw"
          transaction={[Function]}
/>,
        <CryptoPicker
          cryptos={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "userBalance": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "changeBalance": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "btn-deposit",
          "label": "Deposit",
          "transaction": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "btn-withdraw",
          "label": "Withdraw",
          "transaction": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cryptos": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Balance
            userBalance={10}
/>,
          <ChangeBalance
            changeBalance={[Function]}
/>,
          <Button
            className="btn-deposit"
            label="Deposit"
            transaction={[Function]}
/>,
          <Button
            className="btn-withdraw"
            label="Withdraw"
            transaction={[Function]}
/>,
          <CryptoPicker
            cryptos={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "userBalance": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeBalance": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "btn-deposit",
            "label": "Deposit",
            "transaction": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "btn-withdraw",
            "label": "Withdraw",
            "transaction": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cryptos": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
